module/zfs/vdev.c

```diff
@@ -3613,6 +3614,14 @@ vdev_stat_update(zio_t *zio, uint64_t psize)
 			}
 
 			if (zio->io_delta && zio->io_delay) {
+				int n = 1000; // average over 1000 zios
```

Comment:
Well, do you think about a specific usecase where this might need tweaking?

I think 1000 I/Os is a nice value for averaging the delay introduced by compression to be still able to react on different loads on the system while still leveling out short spikes in latency.

But I haven't done any scientific approach to determine the optimal value for this. So this solely rely on guessing.

Reply:
please make it tunable because very large systems can benefit.